WMF Framework/Widget/WidgetController.swift

```diff
@@ -379,4 +383,53 @@ public extension WidgetController {
         }
     }
 
+    // MARK: - Fetch Picture of the Day Widget Content
+
+    func fetchPictureOfTheDayContent(isSnapshot: Bool = false, completion: @escaping (WidgetContentFetcher.PictureOfTheDayResult) -> Void) {
+        func performCompletion(result: WidgetContentFetcher.PictureOfTheDayResult) {
+            DispatchQueue.main.async {
+                completion(result)
+            }
+        }
+
+        let fetcher = WidgetContentFetcher.shared
+        var widgetCache = sharedCache.loadCache()
+
+        guard !isSnapshot else {
+            let previewSnapshot = widgetCache.featuredContent ?? WidgetFeaturedContent.previewContent() ?? WidgetFeaturedContent()
+            if let previewPictureOfTheDay = previewSnapshot.pictureOfTheDay {
+                performCompletion(result: .success(previewPictureOfTheDay))
+            } else {
+                performCompletion(result: .failure(.contentFailure))
+            }
+            return
+        }
+
+        fetchFeaturedContent { result in
+            switch result {
+            case .success(var featuredContent):
+                if let imageSource = featuredContent.pictureOfTheDay?.originalImageSource {
+                    fetcher.fetchImageDataFrom(imageSource: imageSource) { imageResult in
+                        featuredContent.pictureOfTheDay?.originalImageSource?.data = try? imageResult.get()
+                        featuredContent.pictureOfTheDay?.originalImageSource?.source = WMFChangeImageSourceURLSizePrefix(imageSource.source, Int(self.potdTargetImageSize.width))
```

Comment:
Good catch! Oversight on my part. Makes sense that it may fail to fetch a sufficiently large image since it'd be trying to fetch the original highest size/resolution image, not the resized to POTD target image size image. Updated with resizing before the image data fetch. You may have to purge the existing Widget Cache to see the change.

RE: the reassignment, mostly just did that to keep some kind of purity of the data represented in the cache and defensiveness against unexpected reuse of that large sized URL, but I can see how that could be confusing. Happy to remove it if you feel strongly about it.